subsys/dult/motion_detector.c

```diff
@@ -0,0 +1,346 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <stdbool.h>
+#include <errno.h>
+
+#include <zephyr/kernel.h>
+#include <zephyr/random/random.h>
+
+#include "dult.h"
+#include "dult_user.h"
+#include "dult_motion_detector.h"
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(dult_motion_detector, CONFIG_DULT_LOG_LEVEL);
+
+#define SEPARATED_UT_SAMPLING_RATE1	K_SECONDS(10)
+#define SEPARATED_UT_SAMPLING_RATE2	K_MSEC(500)
+#define SEPARATED_UT_BACKOFF_PERIOD	K_HOURS(6)
```

Comment:
It would be great to support the "test" mode that allows the UT Timeout and the UT backoff period to be set to very short intervals. Perhaps we could add a test mode Kconfig? I am open to other ideas if there are other alternatives.

Reply:
Something to keep in mind is that we will need to support runtime change of these parameters for some protocols, but that's an issue for another day